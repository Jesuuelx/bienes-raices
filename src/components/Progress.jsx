import { useSelector } from "react-redux";

export const Progress = () => {
  const { displayName, email, address, floor, aditionalInformation } =
    useSelector((state) => state.steps);

  return (
    <>
      <h1>Proceso Formulario</h1>
      <hr />
      <ol className="mt-5">
        <li>{displayName}</li>
        <li className="mt-5">{email}</li>
        <li className="mt-5">{address}</li>
        <li className="mt-5">{floor}</li>
        <li className="mt-5">
          <ul>
            {" "}
            {aditionalInformation.map((info) => (
              <li key={info.id}>{info.done && info.name} </li>
            ))}{" "}
          </ul>
        </li>
      </ol>
    </>
  );
};
