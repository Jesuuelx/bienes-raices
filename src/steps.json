[
  {
    "nameComponent": "NameStep",
    "name": "Nombre y Apellido",
    "pathTo": "/correo-usuario",
    "id:": 0,
    "description": "¿Quién recibirá la oferta?"
  },
  {
    "nameComponent": "EmailStep",
    "name": "Correo",
    "pathTo": "/direccion",
    "id:": 1,
    "description": "¿A que correo quieres que nos comuniquemos contigo?"
  },
  {
    "nameComponent": "AddressStep",
    "name": "Direccion",
    "pathTo": "/datos-piso",
    "id:": 2,
    "description": "Danos tu direccion"
  },
  {
    "nameComponent": "FloorStep",
    "name": "Piso",
    "pathTo": "/caracteristicas-adicional",
    "id:": 3,
    "description": "Piso de tu apartamento*"
  },
  {
    "nameComponent": "AdditionalStep",
    "name": "Informacion Adicional",
    "pathTo": "/resumen",
    "id:": 4,
    "description": "Elige si es tu caso"
  },
  {
    "nameComponent": "Resumen",
    "name": "Resumen",
    "id:": 5,
    "description": "¡Has completado tu registro con exito!"
  }
]
