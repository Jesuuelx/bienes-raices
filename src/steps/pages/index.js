export * from './AdditionalStep';
export * from './AddressStep';
export * from './EmailStep';
export * from './FloorStep';
export * from './NameStep';
