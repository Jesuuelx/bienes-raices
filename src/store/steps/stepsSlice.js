import { createSlice } from "@reduxjs/toolkit";

export const stepsSlice = createSlice({
  name: "steps",
  initialState: {
    status: "checking",
    displayName: null,
    email: null,
    address: null,
    floor: null,
    aditionalInformation: [
      {
        name: "zonabbq",
        done: false,
        id: 1,
      },
      {
        name: "parque",
        done: false,
        id: 2,
      },
      {
        name: "salondefiesta",
        done: false,
        id: 3,
      },
    ],
    isLoading: false,
    errorMessage: false,
  },
  reducers: {
    stepDisplayName: (state, { payload }) => {
      state.displayName = payload.name;
    },
    stepEmail: (state, { payload }) => {
      state.email = payload.email;
    },
    stepAddress: (state, { payload }) => {
      state.address = payload.address;
    },
    stepFloor: (state, { payload }) => {
      state.floor = payload.floor;
    },
    stepAditionalInformation: (state, { payload }) => {
      const newState = state.aditionalInformation.map((info) =>
        info.id === payload ? { ...info, done: !info.done } : info
      );
      state.aditionalInformation = newState;
    },
    startLoading: (state) => {
      state.isLoading = true;
    },
  },
});
export const {
  stepDisplayName,
  stepEmail,
  stepAddress,
  stepFloor,
  stepAditionalInformation,
  startLoading,
} = stepsSlice.actions;
