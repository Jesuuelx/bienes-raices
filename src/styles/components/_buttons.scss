.btn {
  background-color: transparent;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 12px;
  padding: 7px 10px 7px 10px;

  transition: color 0.3s ease;

  &:focus {
    outline: none;
  }

  &:hover {
    color: darken($color: white, $amount: 10);
  }
}

.btn-primary {
  background-color: $primary;
  border-radius: 8px;
  width: 10rem;
  height:2rem;
 margin-top: 5rem;

  &:disabled {
    background-color: lighten($color: $primary, $amount: 15);
  }

  &:hover {
    background-color: darken($color: $primary, $amount: 15);
  }
}

.btn-danger {
  background-color: red;

  transition: background-color 0.3s ease;

  &:hover {
    background-color: darken($color: red, $amount: 10);
  }
}

.btn-block {
  width: 100%;
}

.btn-modal {

  
  background-color: $primary;
  border-radius: 8px;
  width: 10rem;
  height:2rem;
 margin-top: 5rem;
 display: none;
 color: white;
 border: none;
 font-weight: bold;

  &:disabled {
    background-color: lighten($color: $primary, $amount: 15);
  }

  &:hover {
    background-color: darken($color: $primary, $amount: 15);
  }

  

    
  }

  
  @media screen and (max-width:768px) {
    .btn-modal {

      display:block;

    }
    
  }


