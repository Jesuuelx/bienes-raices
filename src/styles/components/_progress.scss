 .progress__container {
    color: black;
    font-weight: bold;
  } 
  
  .progress__step {
    
    padding: 10px;
    
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    
    background-color: cream;
  }
  
  .progress__v-stepper {
    position: relative;
  /*   visibility: visible; */
  }
  
  
  /* regular step */
  .progress__step .progress__circle {
    background-color: white;
    border: 3px solid gray;
    border-radius: 100%;
    width: 20px;    /* +6 for border */
    height: 20px;
    display: inline-block;
  }
  
  .progress__step .progress__line {
      top: 23px;
    left: 12px;
  /*   height: 120px; */
    height: 100%;
      
      position: absolute;
      border-left: 3px solid $primary;
  }
  
  .progress__step.progress__completed .progress__circle {
    visibility: visible;
    background-color: $primary;
    border-color: $primary
  }
  
  .progress__step.progress__completed .progress__line {
    border-left: 3px solid $primary;
  }
  
  .progress__step.progress__active .progress__circle {
  visibility: visible;
    border-color: $primary;
  }
  
  .progress__step.progress__empty .progress__circle {
      visibility: hidden;
  }
  
  .progress__step.progress__empty .progress__line {
  /*     visibility: hidden; */
  /*   height: 150%; */
    top: 0;
    height: 150%;
  }
  
  
  .progress__step:last-child .progress__line {
    border-left: 3px solid white;
    z-index: -1; /* behind the circle to completely hide */
  }
  
  .progress__content {
    margin-left: 20px;
    display: inline-block;
  }

  ul {
    text-decoration: none;
    list-style: none;
  }
  