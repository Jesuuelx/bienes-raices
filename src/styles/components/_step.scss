.step__main {
  /*   align-items: center;
  background-color: $primary;
  display: flex;
  justify-content: center;
  margin: 0px;
  width: 100vw; */
  /*     display: grid;
    grid-template-columns: repeat(8, 1fr);
    grid-column-gap: 24px; */
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  max-width: 1400px;
 /*  height: 100vh; */
  width: 100%;
}

.step__box-container1 {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: flex-start;
  background-color: white;
  border-radius: 2px;
  padding: 20px;
}

.step__box-step {
  background-color: $gris;
  border-radius: 2px;
  padding: 20px;
  background-color: rgb(252, 250, 250);
  top: 0;
  right: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.step__title {
  color: darken($color: $primary, $amount: 20);
  font-size: 50px;
  line-height: 48px;
  letter-spacing: -2px;
  margin-bottom: 25px;
  margin-top: 40px;
  font-weight: bold;
}

.step__input {
  color: black;
  border-radius: 8px;
  font-size: 16px;
  margin-bottom: 10px;
  height: 40px;
  width: 20rem;
  font-weight: 900;

  transition: border-radius 0.3s ease;

  &:focus {
    border-radius: 1px solid $primary;
    outline: none;
  }
}

.step__input-direccion {
  color: black;
  border-radius: 8px;
  font-size: 16px;
  margin-bottom: 10px;
  height: 40px;
  width: 6rem;
  font-weight: 900;
  margin-left: 0.4rem;

  transition: border-radius 0.3s ease;

  &:focus {
    border-radius: 1px solid $primary;
    outline: none;
  }
}

.step__input-adicional {
  color: black;
  border-radius: 8px;
  font-size: 16px;
  margin-bottom: 10px;
  height: 30px;
  width: 1rem;
  font-weight: 900;
  margin-left: 0.8rem;

  transition: border-radius 0.3s ease;

  &:focus {
    border-radius: 1px solid $primary;
    outline: none;
  }
}

.step__alert-error {
  background-color: $gris;
  border-radius: 5px;
  color: red;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
  font-size: 16px;
  font-weight: bold;
  font-family: Arial, Helvetica, sans-serif;
}

.step__alert-error-direccion {
  background-color: $gris;
  border-radius: 5px;
  color: red;
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
  font-size: 8px;
  font-weight: bold;
  font-family: Arial, Helvetica, sans-serif;
}

.step__form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 40rem;
}

.step__form-direccion {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  width: 40rem;
}
.step__intro-title {
  font-size: 48px;
  line-height: 48px;
  letter-spacing: -2px;
  margin-bottom: 20px;
}

.step__subtitle {
  color: $primary;
}

.step__bold {
  font-weight: 900;
}
.step__label{
  margin-left: 0.8rem;
  color: black;
  font-weight: bold;
}

.modal { 
  display: none;
}


@media screen and (max-width: 768px) {
  .step-main {
    width: 100%;
    flex-direction: column;
    flex-wrap: wrap;
  }
  .step__box-step {
    display: none;
  }

  .step__form {
    width: 100%;
  }

  .step__intro-title {
    font-size: 28px;
    line-height: 48px;
    letter-spacing: -2px;
    margin-bottom: 20px;
  }

  .step__form-direccion {
    flex-direction: column;
    width: 25rem;
  }

  .step__input-direccion {
    width:15rem ;


  }

  .modal {
    position: fixed;
    z-index: 999;
    top: 0;
    left: 0;
    width: 100%;
    min-height: 100vh;
    background-color: rgba(0, 0, 0, 0.75);
    display: none;
    justify-content: center;
    flex-direction: colum;
    align-items: center;
  }
  
  .modal-container {
    position: relative;
    padding: 1rem;
    overflow-y: auto;
    background-color: #fff;
  }
  
  .modal-close {
    position: absolute;
    top: 1rem;
    right: 1rem;
  }
  
  .modal.is-open {
    display: flex;
    justify-content: center;
    flex-direction: colum;
    align-items: center;
  }

}

@media screen and (max-width: 992px) { 
  .step__form-direccion {
    flex-direction: column;
    width: 25rem;
  }

  .step__input-direccion {
    width:20rem ;


  }

  .step__box-container1 {
    align-items: center;
  }

}
