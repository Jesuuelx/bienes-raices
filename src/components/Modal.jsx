import React from "react";
import { Progress } from "./Progress";

export const Modal = ({closeModal, isOpen}) => {

  return (
    <div className={`modal ${isOpen && 'is-open'} `}>
      <div className="modal-container">
        <button className="modal-close" onClick={closeModal}>
          {" "}
          X{" "}
        </button>
        <Progress />
      </div>
    </div>
  );
};
