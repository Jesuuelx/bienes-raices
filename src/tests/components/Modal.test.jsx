import { render } from "@testing-library/react";
import { Modal } from "../../components/Modal";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { configureStore } from "@reduxjs/toolkit";
import { stepsSlice } from "../../store/steps/stepsSlice";


const store = configureStore({
    reducer: {
      steps: stepsSlice.reducer,
    },
    /*    preloadedState: {
            steps:
        } */
  });
describe('Test in Modal', () => { 

    const closeModal = jest.fn
    const isOpen = false;

    test('show Snapchot', () => { 

        const { container } = render(
           
           <Provider store={store}>

            <MemoryRouter>

               <Modal closeModal={closeModal} isOpen={isOpen} />

            </MemoryRouter>

           </Provider>
        )

        expect( container ).toMatchSnapshot();


     })


 })