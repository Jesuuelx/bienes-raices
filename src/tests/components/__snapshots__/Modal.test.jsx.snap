// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in Modal show Snapchot 1`] = `
<div>
  <div
    class="modal false"
  >
    <div
      class="modal-container"
    >
      <button
        class="modal-close"
      >
         
        X
         
      </button>
      <div
        class="progress__container"
      >
        <h1>
          Proceso Formulario:
        </h1>
        <hr />
        <br />
        <div
          class="progress__step false false"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          >
            Nombre:
          </div>
        </div>
        <div
          class="progress__step progress__empty"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          />
        </div>
        <div
          class="progress__step false false"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          >
            Email:
          </div>
        </div>
        <div
          class="progress__step progress__empty"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          />
        </div>
        <div
          class="progress__step false false"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          >
            Direccion:
          </div>
        </div>
        <div
          class="progress__step progress__empty"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          />
        </div>
        <div
          class="progress__step false false"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          >
            Piso
          </div>
        </div>
        <div
          class="progress__step progress__empty"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          />
        </div>
        <div
          class="progress__step false false"
        >
          <div
            class="progress__v-stepper"
          >
            <div
              class="progress__circle"
            />
            <div
              class="progress__line"
            />
          </div>
          <div
            class="progress__content"
          >
            Caracteristicas
            <ul>
              <li />
              <li />
              <li />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
